commit 895ae5bee159d148bac21a82899292c408c1aab1
Author: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>
Date:   Fri Nov 25 11:11:13 2022 +0530

    drivers: misc: Add Support for TMR Inject IP
    
    The Triple Modular Redundancy(TMR) provides functional fault injection by
    changing selected MicroBlaze instructions, which provides the possibility
    to verify that the TMR subsystem error detection and fault recovery logic
    is working properly.
    
    Usage:
    echo 1 > /sys/kernel/debug/xtmr_inject/inject_fault/inject_fault
    
    Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>
    Link: https://lore.kernel.org/r/20221125054113.122833-5-appana.durga.kedareswara.rao@amd.com
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index 3afaf0b50854..6a514fb7a1f2 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -528,6 +528,16 @@ config TMR_MANAGER
 
 	  Say N here unless you know what you are doing.
 
+config TMR_INJECT
+	tristate "Select TMR Inject"
+	depends on TMR_MANAGER && FAULT_INJECTION_DEBUG_FS
+	help
+	  This option enables the driver developed for TMR Inject.
+	  The Triple Modular Redundancy(TMR) Inject provides
+	  fault injection.
+
+	  Say N here unless you know what you are doing.
+
 source "drivers/misc/c2port/Kconfig"
 source "drivers/misc/eeprom/Kconfig"
 source "drivers/misc/cb710/Kconfig"